osu.Game/OsuGame.cs

```diff
@@ -1093,6 +1091,24 @@ private void screenPushed(IScreen lastScreen, IScreen newScreen)
         {
             ScreenChanged(lastScreen, newScreen);
             Logger.Log($"Screen changed → {newScreen}");
+
+            // set AllowTrackAdjustments if new screen defines it, inherit otherwise
+            if (newScreen is IOsuScreen newOsuScreen && newOsuScreen.AllowTrackAdjustments.HasValue)
+            {
+                allowTrackAdjustmentsStack.Push(newOsuScreen.AllowTrackAdjustments.Value);
+                Logger.Log($"Screen's AllowTrackAdjustments explicit → {allowTrackAdjustmentsStack.First()}");
+            }
+            else if (allowTrackAdjustmentsStack.Any())
+            {
+                allowTrackAdjustmentsStack.Push(allowTrackAdjustmentsStack.First());
+                Logger.Log($"Screen's AllowTrackAdjustments inherit → {allowTrackAdjustmentsStack.First()}");
+            }
+            else
+            {
+                allowTrackAdjustmentsStack.Push(false);
```

Comment:
not sure about pushing `false` as the absolute fallback - the old `OsuScreen` default was `true`